dask/dataframe/io/io.py

```diff
@@ -213,6 +230,13 @@ def from_pandas(data, npartitions=None, chunksize=None, sort=True, name=None):
     return new_dd_object(dsk, name, data, divisions)
 
 
+def _infer_npartitions(data, chunksize):
```

Comment:
Does this induce a `compute` on `data`?

Reply:
it uses `memory_usage`that below runs a map partition, so there isn't any explicit `compute`.